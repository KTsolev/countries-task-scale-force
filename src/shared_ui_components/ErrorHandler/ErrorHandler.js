import React from 'react';

export const ErrorHandler = (props) => {
    const { title, onPress } = props;

    return (
        <section className='error-page'>
            <h2>{title}</h2>
            {onPress && <span className='error-button' onClick={onPress} >Try again.</span>}
        </section>
    )
}