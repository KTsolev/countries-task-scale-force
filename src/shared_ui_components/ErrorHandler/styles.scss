.error {

    &-page {
        @include column;
        justify-content: center;
        align-items: center;

        &-button {
            text-align: center;
            width: 120px;
            height: 40px;
            padding: 8px 10px;
            background-color: #ce6f30;
            color: #FFF;
        }
    }

}