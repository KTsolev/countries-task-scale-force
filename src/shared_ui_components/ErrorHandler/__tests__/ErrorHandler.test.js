import React from 'react';
import { ErrorHandler } from '../ErrorHandler';
import renderer, { create, act } from 'react-test-renderer';

const onPress = jest.fn();

describe('ErrorHandler tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ErrorHandler />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when all props are passed', () => {
    const tree = renderer
      .create(<ErrorHandler onPress={onPress} title='Something went wrong!' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should handle onclick', () => {
    const tree = create(<ErrorHandler onPress={onPress} title='Something went wrong!' />).root;

    const button = tree.findAllByType('span')[0];
    
    act(button.props.onClick);

    expect(onPress).toHaveBeenCalled();
  });
})
