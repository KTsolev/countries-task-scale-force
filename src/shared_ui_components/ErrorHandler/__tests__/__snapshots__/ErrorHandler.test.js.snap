// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorHandler tests renders correctly 1`] = `
<section
  className="error-page"
>
  <h2 />
</section>
`;

exports[`ErrorHandler tests when all props are passed 1`] = `
<section
  className="error-page"
>
  <h2>
    Something went wrong!
  </h2>
  <span
    className="error-button"
    onClick={[MockFunction]}
  >
    Try again.
  </span>
</section>
`;
